lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -94,9 +96,9 @@ def self.expose(*args)
           return options[:override_return] if options.key?(:override_return)
           reflection = @object.class.reflection_with_virtual(method)
           if reflection && reflection.collection?
-            wrap_results(ret.to_a)
+            options[:association] ? wrap_results(self.class.filter_objects(ret.to_a)) : wrap_results(ret.to_a)
           else
-            wrap_results(ret)
+            options[:association] ? wrap_results(self.class.filter_objects(ret)) : wrap_results(ret)
```

Comment:
There are other places that pass an association into `wrap_results` e.g.: [here](https://github.com/ManageIq/manageiq/blob/master/lib/miq_automation_engine/service_models/miq_ae_service_classification.rb#L11)

Would it make sense to change `wrap_results` to support a collection coming in and do the to_a there?
In the near term, I want rbac to support collections.